docs/architecture.md

```diff
@@ -0,0 +1,135 @@
+# Architecture
+
+This document provides an overview of the architecture and design principles of the NGINX Kubernetes Gateway. The target
+audience includes the following groups:
+
+* *Cluster Operators* who would like to know how the software works and also better understand how it can fail.
+* *Developers* who would like to [contribute][contribute] to the project.
+
+We assume that the reader is familiar with core Kubernetes concepts, such as Pod, Deployment, Service, and Endpoints.
+Additionally, we recommend reading [this blog post][blog] for an overview of the NGINX architecture.
+
+[contribute]: https://github.com/nginxinc/nginx-kubernetes-gateway/blob/main/CONTRIBUTING.md
+
+[blog]: https://www.nginx.com/blog/inside-nginx-how-we-designed-for-performance-scale/
+
+## What is NGINX Kubernetes Gateway?
+
+The NGINX Kubernetes Gateway is a component in a Kubernetes cluster that configures an HTTP load balancer according to
```

Comment:
Couldn't this technically be grpc, generic TLS, or UDP as well?

Reply:
This refers to the load balancer that exposes the applications in the cluster—the ingress point, which is HTTP(s). The application backends could be gRPC, TCP, or UDP, but the client will always access the applications over HTTP(s).